test/integration/bitbar/serverspec/bitbar_devicepool_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'spec_helper.rb'
+
+describe 'users' do
+  describe user('bclary') do
+    it { should exist }
+  end
+
+  describe user('aerickson') do
+    it { should exist }
+  end
+
+  describe user('jwatkins') do
+    it { should exist }
+  end
+
+  describe user('dhouse') do
+    it { should exist }
+  end
+
+  describe user('bitbar') do
+    it { should exist }
+  end
+end
+
+describe 'git repo' do
+  describe command('cd /home/bitbar/mozilla-bitbar-devicepool && git status') do
+    its(:exit_status) { should eq 0 }
+  end
+end
+
+describe 'service' do
+  # service:is_installed is windows only
+  describe command('systemctl status bitbar') do
+    # code 3 is loaded, but not running
+    its(:exit_status) { should eq 3 }
```

Comment:
This is great; I like having tests! I'm always needing to check state manually when making changes in puppet, and putting those tests into reproducible tests is excellent.

Reply:
Yeah, the cycle of converge and manual check things out gets old. :)